content/en/blog/2023-06-23-unikraft-gsoc-arm-cca-1.md

```diff
@@ -0,0 +1,80 @@
+I am really happy to work on this project.
+
+## Project Overview
+
+### Arm CCA
+
+Arm CCA introduces Realm Management Extension (RME), which extends Arm TrustZone technology with two new security states: `realm` state and `root` state.
+The following figure shows the system architecture of the CCA.
+Instead of running a VM in the normal world, the CCA can run a VM in the realm state.
+The realm state constructs protected execution environments called realms, which protect the data in the realms from other components.
+This architecture allows the hypervisor to control the VM but removes the right for access to that VM.
+The CCA achieves this separation through a combination of architectural hardware extensions (RME) and firmware (Realm Manager in EL2 and Monitor in EL3).
+
+![Arm CCA architecture](https://www.arm.com/-/media/global/Why%20Arm/architecture/security/cca/arm-cca-realms-diagram.png?revision=ea88f294-2e89-49ca-9ecc-9400f7177692&h=650&w=1093&rev=ea88f2942e8949ca9ecc9400f7177692&hash=BE1FD0370318E58F091A56CD689A07B8970F8F9D")
+
+The Realm Management Monitor (RMM) is the realm world firmware that is used to manage the execution of the realm VMs and their interaction with the hypervisor in the normal world.
+The RMM operates in Exception level 2 in the realm world.
+The RMM provides services to the realms, using Realm Service Interface (RSI).
+Therefore, to use Unikraft as an application in a realm, it needs to support the use of RSI.
+Besides, Unikraft also needs to support memory management and attestation in Arm CCA.
+As a reference, Arm has provided an [integration stack](https://gitlab.arm.com/arm-reference-solutions/arm-reference-solutions-docs/-/blob/master/docs/aemfvp-a-rme/user-guide.rst) for Arm's reference CCA software architecture and implementation.
+
+### Objectives
+
+To make Unikraft run on the CCA, we need to achieve the following objectives:
+
+- Add support for kvmtool to run `app-helloworld` in the normal world.
+- Add options for CCA and basic support like RSI commands to run `app-helloworld` in a realm.
+- Prepare the FVP environment and add support for it.
+- Add support for more advanced features like attestation and memory encryption
+- Add testing applications and evaluate the overall project.
+
+## Current Progress
+
+All the progress made and the relevant knowledge of this project is maintained on this [GitHub Project](https://github.com/orgs/unikraft/projects/39/views/1) page.
+My progress in the first three weeks of GSoC can be divided into information gathering and code implementation.
+
+On the information-gathering side, the following questions are answered:
+
+- How unikraft currently boots?
+- How Linux is booted in kvmtool?
+- What devices are supported in kvmtool?
+- What are the changes by Arm to kvmtool for CCA?
+- What are the changes in the realm linux?
+
+On the code-implementation side, some changes are made to Unikraft to make `app-helloworld` run on kvmtool.
```

Comment:
It would be great if you could link the commits relevant to the work you're describing here, if you have any. Otherwise, this looks good.